DigitalLearningSolutions.Web/ViewModels/TrackingSystem/CourseSetup/CourseDetails/EditLearningPathwayDefaultsViewModel.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem.CourseSetup.CourseDetails
+{
+    using DigitalLearningSolutions.Data.Models.Courses;
+
+    public class EditLearningPathwayDefaultsViewModel
+    {
+        public EditLearningPathwayDefaultsViewModel() { }
+
+        public EditLearningPathwayDefaultsViewModel(CourseDetails courseDetails)
+        {
+            CustomisationId = courseDetails.CustomisationId;
+            CompleteWithinMonths = courseDetails.CompleteWithinMonths.ToString();
+            ValidityMonths = courseDetails.ValidityMonths.ToString();
+            Mandatory = courseDetails.Mandatory;
+            AutoRefresh = courseDetails.AutoRefresh;
+        }
+
+        public EditLearningPathwayDefaultsViewModel(
+            int customisationId,
+            string? completeWithinMonths,
+            string? validityMonths,
+            bool mandatory,
+            bool autoRefresh
+        )
+        {
+            CustomisationId = customisationId;
+            CompleteWithinMonths = completeWithinMonths;
+            ValidityMonths = validityMonths;
+            Mandatory = mandatory;
+            AutoRefresh = autoRefresh;
+        }
+
+        public int CustomisationId { get; set; }
+
+        public string? CompleteWithinMonths { get; set; }

```

Comment:
Ah, I had some issues with making these integers because of some accessibility stuff around number inputs. If the type is int and the user enters a a value with letters, it brings up a default error message, something like "Value is not valid for CompleteWithinMonths" (which isn't helpful to the user), that I couldn't manage to get rid of, and doesn't show the error messages from the validation attribute. Let me know if you know a way around that or if you think it's fine to keep it as strings.

Reply:
Created a Numeric Input view component. It's pretty much the same as the Text input view component with configurable model, just has an inputmode="numeric" on it, but definitely makes it clearer in the html clearer that it's a number input rather than text. Couldn't apply the property pattern="[0-9]*" which is on the date component because it comes with a built-in pop up error message if the user enters letters which isn't very helpful or accessible, and blocks the error message from the validation attribute.

Reply:
Ah, I've added novalidate to the form and now the validation works properly with pattern="[0-9]*" added to the view component. Thanks for that!